base/ltfsstrc.dtx

```diff
@@ -1165,6 +1168,45 @@
 %    \begin{macrocode}
 %<+debug> \poptracing
     }
+%    \end{macrocode}
+% \end{macro}
+%
+% \begin{macro}{\transform@scriptfont}
+%    |\transform@scriptfont| has two arguments: the name of the script size we transform
+%    the font for (always |sf| or |ssf|) and the \emph{family/series/shape\/} name as a csname.
+%    It expands to a new \emph{family/series/shape\/} name which will be used for this
+%    script.
+%    For this |\transform@scriptfont| mostly delegates to |\__nfss_transform_scriptfont:nw|, which
+%    searches for a replacement marker
+%    |\__nfss_mapped_scriptfont_family_|\meta{scriptsize}|_|\meta{encoding}|/|\meta{family}
+%    with a remapped encoding and family.
+%    \begin{macrocode}
+\ExplSyntaxOn
+\cs_set:Npn \transform@scriptfont #1 #2 {
+  \exp_last_unbraced:Nno
+    \__nfss_transform_scriptfont:nw
+    {#1}
+    {\token_to_str:N #2}
+}
+\cs_set:Npn \__nfss_transform_scriptfont:nw #1 #2 / #3 / {
+  \cs_if_exist_use:cF { __nfss_mapped_scriptfont_family_ #1 _ #2 / #3 } { #2 / #3 }
+  /
+}
+%    \end{macrocode}
+% \end{macro}
+%
+% \begin{macro}{\declare@scriptfont@mapping}
+%    |\declare@scriptfont@mapping| can be used to define replacement font families which will be used
+%    for math subscripts for subsubscripts when a specific font family is used for the main font.
+%    For this |\declare@scriptfont@mapping| takes five arguments: The first argument should be \texttt{sf}
+%    \texttt{ssf} to indicate if the replacement should be used for the scriptfont or the scriptscriptfont.
+%    Then  it expects the encoding and family name of the base font for which the mapping should be applied
+%    and the encoding and family of the font which will be used for the scriptfont.
+%    \begin{macrocode}
```

Comment:
not sure why this should be an internal 2e command. It seems to me rather a font setup declaration as any of the other ones, so should be \DeclareScriptfontMapping or something like that (one would need to check what others are around to make sure it fits.

Where should these declarations live? In font support packages? In the kernel? ...?

Reply:
Currently all expected uses are in `unicode-math` (or `lua-unicode-math`) but in theory they could also live in font support packages,

Reply:
The mappings should be created by the font author in their font support package.

Reply:
`\DeclareScriptfontMapping` would be fine for me. I couldn't find any existing commands describing a mapping between different font families and no mention of script fonts in command names either, so there is not too much concern about consistency. Existing commands dealing with script fonts refer to MathSizes, but that wouldn't fit here since it explicitly is not about sizes.

Reply:
Its the consistency with other NFSS related commands .. but too late for me to review that tonight

Reply:
> The mappings should be created by the font author in their font support package.

then it should be CamelCase I would say

Reply:
> Its the consistency with other NFSS related commands .. but too late for me to review that tonight

it belongs next to \DeclareMathSizes I would say, so `\DeclareMath<something>`